src/utils/formik-helpers.ts

```diff
@@ -112,26 +113,65 @@ export const touchedDescriptionTabFields: FormikTouched<Publication> = {
   },
 };
 
-export const touchedReferenceTabFields: FormikTouched<Publication> = {
-  entityDescription: {
-    reference: {
-      publicationContext: {
-        type: true,
-        title: true,
-      },
-      publicationInstance: {
-        type: true,
-        articleNumber: true,
-        issue: true,
-        // pages: {
-        //   begin: true,
-        //   end: true,
-        // },
-        peerReviewed: true,
-        volume: true,
-      },
-    },
-  },
+export const touchedReferenceTabFields = (publicationType: PublicationType | ''): FormikTouched<any> => {
+  switch (publicationType) {
+    case PublicationType.PUBLICATION_IN_JOURNAL:
+      return {
+        entityDescription: {
+          reference: {
+            publicationContext: {
+              type: true,
+              title: true,
+            },
+            publicationInstance: {
+              type: true,
+              articleNumber: true,
+              issue: true,
+              pages: {
+                begin: true,
+                end: true,
+              },
+              peerReviewed: true,
+              volume: true,
+            },
+          },
+        },
+      };
+
+    case PublicationType.DEGREE:
+      return {
+        entityDescription: {
+          reference: {
+            publicationContext: {
+              type: true,
+              title: true,
+            },
+          },
+        },
+      };
+    default:
```

Comment:
hadde kanskje foretrukket å bruke typene her også i stedet for default. Hvis man ikke har valgt type enda vil vi vel egentlig ikke sette feks issue til touched å gi feilmelding på det

Reply:
jaja, jeg er helt enig med deg. Men jeg vil ta en og en type. Det blir for mye rot hvis jeg skal ta alt i ett.